### ISSUE TITLE
[FEATURE]  Make configurable A or B button to validate in ES menu

### ISSUE THREAD

**Issue by air-sky-one** [recalbox/recalbox-os#863]
### What happened
Since V4 of Recalbox, A button for validation in ES menu has be changed to B button. It can be in some case frustrating for end users, espacially when they go into retrorach menu during playing where the A button is still used for validation.

### What could be great
Make the validation button in ES configurable ! :)

**Comment by menelkir** [comment]
+1

On Mon, Apr 4, 2016 at 9:22 AM, air-sky-one <notifications@github.com>
wrote:

> What happened
>
> Since V4 of Recalbox, A button for validation in ES menu has be changed to
> B button. It can be in some case frustrating for end users, espacially when
> they go into retrorach menu during playing where the A button is still used
> for validation.
> What could be great
>
> Make the validation button in ES configurable ! :)
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/recalbox/recalbox-os/issues/863>
>



-- 
Daniel de Oliveira

**Comment by nadenislamarre** [comment]
the thing is more to reverse the order in retroarch to homogeneize them.

**Comment by digitalLumberjack** [comment]
@nadenislamarre i don't think it worth patching retroarch, as many casual gamers do not even open this menu once. 
This switch would be useful for me too as i prefere O to select and X to cancel :P

**Comment by nadenislamarre** [comment]
who said patching ?
it is customizable.

@rockaddicted, can i let you this task ? i guess it's in the retroarch configgen.

# menu_ok_btn          =
# menu_cancel_btn      =

**Comment by rockaddicted** [comment]
yep, indeed. So in a first time we can force the preset to be coordinated with current ES order (B validate / A return). 
Then in a second time, when this feature will be available in ES , (in 4.1 ?), add a switch to the configgen, to use the same preset in retroach+ES.

**Comment by digitalLumberjack** [comment]
I didn't saw those options. And 1 hour ago you didn't know too :P @nadenislamarre

**Comment by nadenislamarre** [comment]
but 55 minutes ago, yes.

**Comment by rockaddicted** [comment]
hummm don't seem to work.
In my retroarch config file, I have that : 
```
input_menu_ok_btn = 0
input_menu_cancel_btn = 1
```
On my gamepad, `A = 1` and `B = 0`

But once in retroarch menu, `A = ok` and `B = cancel`

**Comment by digitalLumberjack** [comment]
Open the Retroarch CFG file, look for menu_ok and menu_cancel and invert both values.
That's what i found, can you try with those values ?

**Comment by digitalLumberjack** [comment]
Where did you find the infos @nadenislamarre ?
https://www.google.fr/webhp?q=%22input_menu_cancel_btn%22+retroarch#safe=off&q=%22input_menu_cancel_btn%22+retroarch

**Comment by rockaddicted** [comment]
Tried : 
```
input_menu_ok_btn = 0
input_menu_cancel_btn = 1
```
and
```
input_menu_ok_btn = 1
input_menu_cancel_btn = 0
```
And both, in retroarch menu, `A = ok` and` B = cancel`

**Comment by digitalLumberjack** [comment]
`menu_ok=1`
 and `menu_cancel=1` ?

**Comment by rockaddicted** [comment]
So with : 
```
menu_ok_btn = 0
menu_cancel_btn = 
```
Buttons in retroarch switchs ... But not on `A = cancel` and `B = ok` as should be, but as `Y = cancel` and `B = ok`

In my gamepad setup, `Y = 2`

**Comment by rockaddicted** [comment]
So with : 
```
menu_ok_btn = 0
menu_cancel_btn = 1
```
Buttons in retroarch switchs ... But not on `A = cancel` and `B = ok` as should be, but as `Y = cancel` and `B = ok`

In my gamepad setup, `Y = 2`

**Comment by nadenislamarre** [comment]
So, does it work finally ?
i'll give you the retroarch source grep command if needed ;-)

**Comment by nadenislamarre** [comment]
or simplyly :
https://github.com/libretro/RetroArch/blob/master/retroarch.cfg

**Comment by rockaddicted** [comment]
nope in my retroarch config file, I have `menu_cancel_btn = 1` so that should use the `A` to cancel, but it is using the `Y` button on my gamepad.

**Comment by jesus2099** [comment]
I had the same problem as @air-sky-one (original poster) and thought it should be treated over there at ES (Aloshi/EmulationStation#627).

The original issue is that ES has reverse button layout while retroarch and games are OK but you are planning to spread the unwanted button mapping further until some way of swapping is then eventually found, that’s a temporary regression.

https://github.com/recalbox/recalbox-os/issues/863#issuecomment-205460383 by @rockaddicted:

> So in a first time we can force the preset to be coordinated with current ES order (B validate / A return).
> Then in a second time, when this feature will be available in ES , (in 4.1 ?), add a switch to the configgen, to use the same preset in retroach+ES.

:-1: I do appreciate your efforts but please, if possible, wait for the ticket to be fixable before introducing the unwanted behaviour in everywhere instead of just the ES menu. :)
